src/runmode-dpdk.c

```diff
@@ -1102,6 +1130,87 @@ static int DeviceConfigureIPS(DPDKIfaceConfig *iconf)
     SCReturnInt(0);
 }
 
+static void DeviceRingNameInit(char *name_out, const char *format, uint16_t q_num)
+{
+    char name[RTE_RING_NAMESIZE];
+    char q_num_str[5];
+    char *q_pos;
+
+    if (q_num > 9999)
+        FatalError(SC_ERR_DPDK_CONF, "Number of queues must not exceed 10000 queues per interface");
+    q_pos = strstr(format, DPDK_CONFIG_DEFAULT_QUEUE_NUM_SPECIFIER);
+    snprintf(name, q_pos - format + 1, "%s", format);
+    snprintf(q_num_str, sizeof(q_num_str), "%" PRIu16, q_num);
+    strlcat(name, q_num_str, strlen(q_num_str));
+    // copy the rest after queue number specifier
+    strlcat(name, q_pos + strlen(DPDK_CONFIG_DEFAULT_QUEUE_NUM_SPECIFIER),
+            sizeof(name) - strlen(name) - 1);
+    strlcpy(name_out, name, sizeof(name));
+}
+
+static bool DeviceRingNameIsValid(const char *name)
+{
+    unsigned long len = strlen(name);
+    if (len > RTE_RING_NAMESIZE) {
+        SCLogError(SC_ERR_DPDK_CONF, "Ring name (entry \"interface\" %s) cannot be longer than %lu",
+                name, RTE_RING_NAMESIZE);
+        return false;
+    } else if (strstr(name, DPDK_CONFIG_DEFAULT_QUEUE_NUM_SPECIFIER) == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF,
+                "Ring name (entry \"interface\" %s) omits the queue number specifier - \"%s\"",
+                name, DPDK_CONFIG_DEFAULT_QUEUE_NUM_SPECIFIER);
+        return false;
+    }
+    return true;
+}
+
+static int DevicePrefilterAttach(DPDKIfaceConfig *iconf)
+{
+    SCEnter();
+    iconf->rx_rings = SCCalloc(iconf->threads, sizeof(struct rte_ring *));
+    if (iconf->rx_rings == NULL) {
+        SCLogError(SC_ERR_DPDK_INIT, "Failed to calloc rx rings");
+        SCReturnInt(-ENOMEM);
+    }
+
+    iconf->tx_rings = SCCalloc(iconf->threads, sizeof(struct rte_ring *));
+    if (iconf->tx_rings == NULL) {
+        SCLogError(SC_ERR_DPDK_INIT, "Failed to calloc tx rings");
+        SCReturnInt(-ENOMEM);
+    }
+
+    if (!DeviceRingNameIsValid(iconf->iface))
+        SCReturnInt(-EINVAL);
+
+    for (int32_t i = 0; i < iconf->threads; ++i) {
+        char name[RTE_RING_NAMESIZE];
+        DeviceRingNameInit(name, iconf->iface, i);
+        SCLogInfo("Looking up rx ring: %s", name);
+        iconf->rx_rings[i] = rte_ring_lookup(name);
+        if (iconf->rx_rings[i] == NULL) {
+            SCLogError(SC_ERR_DPDK_INIT, "rte_ring_lookup(): cannot get rx ring '%s'\n", name);
```

Comment:
no `\n`

Reply:
multiple instances of this issue, please review all

Reply:
should be fixed